Mathlib/RingTheory/Valuation/Extension.lean

```diff
@@ -160,6 +158,61 @@ theorem instIsLocalHomValuationInteger {S ΓS : Type*} [CommRing S]
 
 end integer
 
+section AlgebraInstances
+
+open IsLocalRing Valuation ValuationSubring
+
+variable {K L Γ₀ Γ₁ : outParam Type*} [Field K] [Field L] [Algebra K L]
+  [LinearOrderedCommGroupWithZero Γ₀] [LinearOrderedCommGroupWithZero Γ₁] (vK : Valuation K Γ₀)
+   (vL : Valuation L Γ₁) [vK.HasExtension vL]
+
+local notation "K₀" => vK.valuationSubring
+local notation "L₀" => Valuation.valuationSubring vL
+
+lemma algebraMap_mem_valuationSubring (x : K₀) : algebraMap K L x ∈ L₀ := by
+  rw [mem_valuationSubring_iff, ← _root_.map_one vL, ← _root_.map_one (algebraMap K L),
+    val_map_le_iff (vR := vK), _root_.map_one]
+  exact x.2
+
+instance instAlgebra_valuationSubring : Algebra K₀ L₀ :=
+  let f : K₀ →+* L₀ := {
+    toFun := fun x ↦ ⟨algebraMap K L x, algebraMap_mem_valuationSubring vK vL x⟩
+    map_one'  := by simp [_root_.map_one, ← @OneMemClass.coe_eq_one]
+    map_mul'  := by simp [MulMemClass.coe_mul, _root_.map_mul, MulMemClass.mk_mul_mk, implies_true]
+    map_zero' := by simp [← ZeroMemClass.coe_eq_zero, _root_.map_zero]
+    map_add'  := by simp only [AddMemClass.coe_add, _root_.map_add, AddMemClass.mk_add_mk,
+      implies_true] }
+  f.toAlgebra
+
+@[simp]
+lemma coe_algebraMap_valuationSubring_eq (x : K₀) :
+  (algebraMap K₀ L₀ x : L) = algebraMap K L (x : K) := rfl
+
+instance instIsScalarTower_valuationSubring : IsScalarTower K₀ K L := by
+  refine IsScalarTower.of_algebraMap_smul ?_
+  intro k l
+  have hkl : k • l = (k : K) • l := rfl
+  simp [Algebra.smul_def, hkl]
+
+instance instIsScalarTower_valuationSubring' : IsScalarTower K₀ L₀ L := by
+  refine IsScalarTower.of_algebraMap_smul ?_
+  intro k l
+  have hkl : k • l = (k : K) • l := rfl
+  simp [Algebra.smul_def, hkl]
+
+lemma algebraMap_mem_maximalIdeal_iff {x : K₀} :
+    algebraMap K₀ L₀ x ∈ (maximalIdeal L₀) ↔ x ∈ maximalIdeal K₀ := by
+  simp only [mem_maximalIdeal_iff, coe_algebraMap_valuationSubring_eq,
+    val_map_lt_one_iff vK vL]
+
+instance : Algebra (ResidueField K₀) (ResidueField L₀) :=
+  (Ideal.Quotient.lift (maximalIdeal K₀)
+    ((Ideal.Quotient.mk (maximalIdeal L₀)).comp (algebraMap K₀ L₀))
+    (fun _ hx ↦ Ideal.Quotient.eq_zero_iff_mem.mpr
+      ((algebraMap_mem_maximalIdeal_iff vK vL).mpr hx))).toAlgebra
+
```

Comment:
```suggestion
instance : Algebra (ResidueField K₀) (ResidueField L₀) :=
  (Ideal.Quotient.lift (maximalIdeal K₀)
    ((Ideal.Quotient.mk (maximalIdeal L₀)).comp (algebraMap K₀ L₀))
    (fun _ hx ↦ Ideal.Quotient.eq_zero_iff_mem.mpr
      ((algebraMap_mem_maximalIdeal_iff vK vL).mpr hx))).toAlgebra

lemma algebraMap_residue_eq_residue_algebraMap (x : K₀) :
    (algebraMap (ResidueField K₀) (ResidueField L₀)) (IsLocalRing.residue K₀ x) =
      IsLocalRing.residue L₀ (algebraMap K₀ L₀ x) :=
  rfl
```

It's too bad that `IsLocalRing.residue` is noncomputable, could have referred to it directly in the algebra defn. Still, good to give the algebraMap definitional lemma.